datastore-api/app/core/kgs/connector.py

```diff
@@ -362,7 +389,65 @@ async def get_object_by_id_msearch(self, kg_name, ids):
                 logger.info("Not FOUND")
         return objs
 
-    async def extract_subgraph(self, kg_name, nids: List[str], hops=2):
+
+
+    async def extract_extra_nodes(self, kg_name, node_ids):
+        cross_combined_ids = []
+        for s in node_ids:
+            for t in node_ids:
+                cross_combined_ids.append([s, t])
+
+        all_edges = await self.get_edge_msearch(kg_name, cross_combined_ids)
+
+        return all_edges
+
+    async def adjacent_matrix_allinone(self, kg_name, node_ids):
+        start=time.time()
+        id2relation = ['antonym','atlocation','capableof','causes','createdby','isa','desires','hassubevent','partof','hascontext','hasproperty','madeof','notcapableof','notdesires','receivesaction','relatedto','usedfor']
+        relation2id = {r: i for i, r in enumerate(id2relation)}
+        cids = np.array(node_ids, dtype=np.int32)
+        n_rel = len(id2relation)
+        n_node = cids.shape[0]
+        adj = np.zeros((n_rel, n_node, n_node), dtype=np.uint8)
+        cids_stringified = []
+
+        for i in cids:
+            if 'n' in str(i):
+                cids_stringified.append(i)
+            else:
+                cids_stringified.append("n"+str(i))
```

Comment:
We do not need to add "n" to IDs. We can just assume that the correct IDs in string type are given and return corresponding errors if not